webapi/Controllers/DocumentImportController.cs

```diff
@@ -276,16 +292,41 @@ private async Task ValidateDocumentImportFormAsync(DocumentImportForm documentIm
                 case SupportedFileType.Jpg:
                 case SupportedFileType.Png:
                 case SupportedFileType.Tiff:
-                {
-                    if (this._ocrSupportOptions.Type != OcrSupportOptions.OcrSupportType.None)
+                    if (this._ocrSupportOptions.Type == OcrSupportOptions.OcrSupportType.None)
                     {
+                        if (documentImportForm.UseContentSafety)
+                        {
+                            if (!this._contentSafety!.ContentSafetyStatus(this._logger))
+                            {
+                                throw new ArgumentException("Unable to analyze image. Content Safety is currently disabled in the backend.");
+                            }
+
+                            var violations = new List<string>();
+                            try
+                            {
+                                // Convert the form file to a base64 string
+                                var base64Image = await this.ConvertFormFileToBase64Async(formFile);
+
+                                // Call the content safety controller to analyze the image
+                                var imageAnalysisResponse = await this._contentSafety!.ImageAnalysisAsync(base64Image, default);
+                                violations = AzureContentSafety.ParseViolatedCategories(imageAnalysisResponse, this._contentSafety!.Options!.ViolationThreshold);
+                            }
+                            catch (Exception ex) when (!ex.IsCriticalException())
+                            {
+                                this._logger.LogError(ex, "Failed to analyze image {0} with Content Safety. ErrorCode: {{1}}", formFile.FileName, (ex as AIException)?.ErrorCode);
+                                throw new ArgumentException($"Failed to analyze image {formFile.FileName} with Content Safety. {(ex as AIException)?.ErrorCode}");
```

Comment:
If this API call is failing it may be due to an issue on our end, or a network slowdown or something unrelated to the request, in which case a 500-level error would make more sense.